src/libcore/cell.rs

```diff
@@ -1005,7 +1006,9 @@ impl<T: ?Sized> RefCell<T> {
     #[inline]
     pub unsafe fn try_borrow_unguarded(&self) -> Result<&T, BorrowError> {
         if !is_writing(self.borrow.get()) {
-            Ok(&*self.value.get())
+            // SAFETY: `self.value.get()` refers to the value owned by `self`
+            // and is thus guaranteed to be valid.
```

Comment:
This is correct but not "complete", I guess, the real requirements here are something like:

* We check that nobody is actively writing now, but it is the caller's responsibility to ensure that nobody writes until the returned reference is no longer in use.